CRM/Report/Utils/Report.php

```diff
@@ -298,31 +297,22 @@ public static function makeCsv(&$form, &$rows) {
    */
   public static function getInstanceID() {
 
-    $config = CRM_Core_Config::singleton();
-    $arg = explode('/', $_GET[$config->userFrameworkURLVar]);
+    $arg = explode('/', CRM_Utils_System::currentPath());
 
-    if ($arg[1] == 'report' &&
-      CRM_Utils_Array::value(2, $arg) == 'instance'
-    ) {
-      if (CRM_Utils_Rule::positiveInteger($arg[3])) {
-        return $arg[3];
-      }
+    if (isset($arg[3]) && $arg[1] == 'report' && $arg[2] == 'instance' && CRM_Utils_Rule::positiveInteger($arg[3])) {
```

Comment:
@colemanw what if there is no `2` key? I guess that would be covered by the first isset right?

Reply:
Right. It's always going to be a non-associative array, so if `[3]` is set then so are `[2]` and `[1]` and `[0]`.